src/effects/effectprocessor.h

```diff
@@ -108,17 +108,23 @@ class EffectProcessorImpl : public EffectProcessor {
         if (kEffectDebugOutput) {
             qDebug() << "~EffectProcessorImpl" << this;
         }
+        int inputChannelHandleNumber = 0;
         for (ChannelHandleMap<EffectSpecificState*>& outputsMap : m_channelStateMatrix) {
+            int outputChannelHandleNumber = 0;
             for (EffectSpecificState* pState : outputsMap) {
                 VERIFY_OR_DEBUG_ASSERT(pState != nullptr) {
                     continue;
                 }
                 if (kEffectDebugOutput) {
-                    qDebug() << "~EffectProcessorImpl deleting state" << pState;
+                    qDebug() << "~EffectProcessorImpl deleting EffectState" << pState
+                             << "for input ChannelHandle(" << inputChannelHandleNumber << ")"
```

Comment:
I will try changing it to an index based loop. The QVarLengthArray API may make it a bit odd though.

Reply:
The ChannelHandle, ChannelHandleMap, and ChannelHandleFactory APIs do not provide a way to do an index based loop here. It would be possible to add new functionality to ChannelHandleFactory to get the ChannelHandle for a given integer index, but I think that would kinda defeat the purpose of the ChannelHandle abstraction so I'm not sure it is a good idea. On the other hand, this hack relies on implementation details of ChannelHandle and ChannelHandleMap. So either way is kinda ugly IMO.

Reply:
Ok, than keep the current solution.